extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetInstructions.java

```diff
@@ -722,75 +800,22 @@ public Builder(final ParquetInstructions parquetInstructions) {
             indexColumns = readOnlyParquetInstructions.getIndexColumns().orElse(null);
         }
 
-        private void newColumnNameToInstructionsMap() {
-            columnNameToInstructions = new KeyedObjectHashMap<>(new KeyedObjectKey.Basic<>() {
-                @Override
-                public String getKey(@NotNull final ColumnInstructions value) {
-                    return value.getColumnName();
-                }
-            });
-        }
-
-        private void newParquetColumnNameToInstructionsMap() {
-            parquetColumnNameToInstructions =
-                    new KeyedObjectHashMap<>(new KeyedObjectKey.Basic<>() {
-                        @Override
-                        public String getKey(@NotNull final ColumnInstructions value) {
-                            return value.getParquetColumnName();
-                        }
-                    });
-        }
-
         public Builder addColumnNameMapping(final String parquetColumnName, final String columnName) {
-            if (parquetColumnName.equals(columnName)) {
-                return this;
-            }
```

Comment:
Why are we skipping this check?

Reply:
This is arguably incorrect, esp wrt the stateful errors that are being thrown. We don't want to allow the user to do:

```java
.addColumnNameMapping("Parquet1", "Foo")
.addColumnNameMapping("Foo", "Foo") // this should throw error, not silently return
```

Reply:
Makes sense.